src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -46,6 +49,30 @@ internal static void AdvanceStream(Stream archiveStream, long bytesToDiscard)
             }
         }
 
+        // Asynchronously helps advance the stream a total number of bytes larger than int.MaxValue.
+        internal static async ValueTask AdvanceStreamAsync(Stream archiveStream, long bytesToDiscard, CancellationToken cancellationToken)
+        {
+            if (archiveStream.CanSeek)
+            {
+                archiveStream.Position += bytesToDiscard;
+            }
+            else if (bytesToDiscard > 0)
+            {
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(minimumLength: MaxBufferLength);
+                while (bytesToDiscard > 0)
+                {
+                    int currentLengthToRead = (int)Math.Min(MaxBufferLength, bytesToDiscard);
+                    int bytesRead = await archiveStream.ReadAsync(buffer.AsMemory(0, currentLengthToRead), cancellationToken).ConfigureAwait(false);
+                    if (bytesRead != currentLengthToRead)
+                    {
+                        throw new EndOfStreamException();
+                    }
+                    bytesToDiscard -= currentLengthToRead;
```

Comment:
Actually, this can be changed to use the stream APIs @eerhardt added recently: https://github.com/dotnet/runtime/pull/69272/

Although I will have to keep the loop, since `bytesToDiscard` is a `long`.